diff --git a/fresnel/CMakeLists.txt b/fresnel/CMakeLists.txt
index d4a570f..e2b3268 100644
--- a/fresnel/CMakeLists.txt
+++ b/fresnel/CMakeLists.txt
@@ -18,6 +18,10 @@ endif()
 
 pybind11_add_module(_common ${_common_sources})
 target_link_libraries(_common PRIVATE QHull::qhull_r QHull::qhull_cpp)
+if(APPLE)
+    add_custom_command(TARGET _common POST_BUILD
+                       COMMAND install_name_tool ARGS -change lib/libqhull_r.8.0.dylib @rpath/libqhull_r.8.0.dylib $<TARGET_FILE:_common>)
+endif()
 install(TARGETS _common LIBRARY DESTINATION ${PYTHON_SITE_INSTALL_DIR})
 
 # _cpu
